Firmware/util.cpp

```diff
@@ -444,35 +444,23 @@ pStrEnd=strchr(pStrBegin,GCODE_DELIMITER);
 if(!pStrEnd)
      return(NULL);
 *nLength=pStrEnd-pStrBegin;
-return(pStrBegin);
+pStrBegin[*nLength] = '\0';
```

Comment:
Note: This changes make it so that the return string is `MK3SMMU2S` and not `MK3SMMU2S"`